extensions/amp-apester-media/0.1/utils.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2016 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as events from '../../../src/event-helper';
+
+const rules = [
+  // if it says it's a webview, let's go with that
+  'WebView',
+  // iOS webview will be the same as safari but missing "Safari"
+  '(iPhone|iPod|iPad)(?!.*Safari)',
```

Comment:
"Standalone" apps (e.g. web-app enabled and added to homescreen) also miss "Safari" so they may not be very accurate.

We ask webview AMP viewers to tell the AMP page, that's exposed under `viewer.isWebviewEmbedded()` from viewer. You may want to use that instead. Up to you, not sure how much accuracy of this matters to you.

Reply:
The accuracy is enough for our use. So we will keep it that way for now :)